controllers/fenceagentsremediation_controller_test.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2023.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	"github.com/medik8s/fence-agents-remediation/pkg/cli"
+)
+
+const (
+	defaultNamespace = "default"
+)
+
+var (
+	executedCommand []string
+)
+
+var _ = Describe("FAR Controller", func() {
+	var (
+		underTestFAR *v1alpha1.FenceAgentsRemediation
+	)
+
+	Context("Defaults", func() {
+		BeforeEach(func() {
+			underTestFAR = &v1alpha1.FenceAgentsRemediation{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-far", Namespace: defaultNamespace},
+			}
+			err := k8sClient.Create(context.Background(), underTestFAR)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(context.Background(), underTestFAR)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("creating a resource", func() {
+			It("should have a namespace scoped CR", func() {
+				Expect(underTestFAR.Namespace).To(Not(BeEmpty()))
+			})
+		})
+	})
+
+	Context("Reconcilie", func() {
+		//Scenarios
+		nodeName :=  "master-0"
```

Comment:
Nit: even though the name is irrelevant for the test functionality I think that using "master" might be a bit confusing.
I suggest going with the conventional "worker"